### PR TITLE
feat: GaussianMixtureConditional

### PR DESCRIPTION
**Pull request by lin-toto**
#238, Gaussian Mixture Model implementation which is used in Cheng2020 paper. 

Implementation details follow the [official code](https://github.com/ZhengxueCheng/Learned-Image-Compression-with-GMM-and-Attention) but I changed CDF generation into a parallel algorithm so that it's easier to work with parallel context models.

Also modified `rans_interface` to allow a torch tensor to be CDF, or ~80% of execution time is wasted on type conversion.

### CODE REVIEW

**Comment by fracape** [comment]
sorry for the late response, I was busy with other things. Can you give me push access to your fork for a quick fix before merge

**Comment by lin-toto** [comment]
> sorry for the late response, I was busy with other things. Can you [give me push access](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork) to your fork for a quick fix before merge

Thanks, though the "allow edits by maintainers" option seems already on my side. I've also sent an invitation to my fork if this doesn't work.

**Comment by fracape** [comment]
Correct, I messed-up and created a new branch, thanks

**Review by fracape — APPROVED** [review]
partially tested, approving. TODO (fracape): train, test and upload model

File: compressai/entropy_models/entropy_models.py
Comment (lin-toto) [reply compressai/entropy_models/entropy_models.py:711]: Sorry I must have missed this comment somehow and just came across it after 4 months.

I developed this GMM variant for [some other model architecture](https://ieeexplore.ieee.org/abstract/document/10095875/) so haven't tried on the exactly original Cheng2020. 

I do used to work with the lab that published the Cheng2020 paper. Let me go ask a few guys if they have some pretrained models.

**Pull request merged**
